R/drop_name_crossref.R

```diff
@@ -0,0 +1,46 @@
+#' Create visual citation from Crossref data
+#'
+#' Takes one or several dois and extracts information from Crossref, then processes them into visual citations with \code{\link{drop_name}}
+#'
+#' @param dois One or several dois to create visual citations for. If they are named, these names are used as filenames; otherwise they are generated based on authors and years.
+#' @inheritDotParams drop_name -bib -cite_key
+#'
+#' @export
+
+drop_name_crossref <- function(dois, ...) {
+
+  if (!"rcrossref" %in% rownames(installed.packages())) {
+    stop("This function requires the rcrossref package. Please install and retry.")
+  }
+
+  stopifnot(is.character(dois))
+
+  ref <-  rcrossref::cr_cn(dois, "citeproc-json")
+
+  to_df <- function(x) {
+    authors <- dplyr::pull(dplyr::mutate(x$author, name = paste0(family, ", ", given)))
+    author_last <- x$author$family[1]
+    dplyr::tibble(YEAR = x$created$`date-parts`[1], JOURNAL = x$`container-title`, AUTHOR = list(authors),
+                  TITLE = x$title, BIBTEXKEY = paste(author_last, x$created$`date-parts`[1]), DOI = x$DOI)
```

Comment:
I would prefer either a paste0 at this point or specifying a separator like _ to prevent filenames with spaces.

Reply:
Makes sense. However, `author_last` will often contain spaces as well - so should we rather wrap the whole key into stringr::str_replace(___, " ", "_")?

Reply:
that sounds good.